pkgs/development/tools/language-servers/fish-lsp/default.nix

```diff
@@ -0,0 +1,85 @@
+{
+  lib,
+  stdenv,
+  fetchYarnDeps,
+  fixup-yarn-lock,
+  fetchFromGitHub,
+  yarn,
+  fish,
+  nodejs,
+}: let
+  version = "1.0.6";
+
+  src = fetchFromGitHub {
+    owner = "ndonfris";
+    repo = "fish-lsp";
+    rev = version;
+    hash = "sha256-o5WomHED/JaamHkEOHdRJHBkfbrfJ4NbvnD/1q1yYSs=";
+  };
+
+  yarnDeps = fetchYarnDeps {
+    yarnLock = "${src}/yarn.lock";
+    hash = "sha256-hmaLWO1Sj+2VujrGD2A+COfVE2D+tCnxyojjq1512K4=";
+  };
+in
+  stdenv.mkDerivation
```

Comment:
Why not using `mkYarnPackage` ?

Reply:
I got some reason to why not using `mkYarnPackage`. 

First the documentation about `mkYarnPackage` is none.

Second with some scraping of existing `mkYarnPackage` I still have issue in my packing since by default I didn't have in my derivation, the source. They was in a `deps/{pname}` directory.

Third, even when I'm switching to this directory, the build script for the wasm made by the author was failing to work since it's try to find a file in the `node_modules` of the directory and not the parent `node_modules`.

So in my search of usage for `mkYarnPackage` I stumble across the packaging of `pkgs/tools/networking/ratman/` which directly use `fetchYarnDeps` and configure yarn to create a `node_modules` directly in the right directory.

Also `stdenv.mkDerivation` directly copy to the source directory the `src`.